### PR TITLE
Fixes #15187: Expose content comparison for all content types

### PR DESCRIPTION
**Pull request by johnpmitsch**
This adds content comparison for content views to the API for packages, puppet modules in addition to errata.

### CODE REVIEW

**Comment by johnpmitsch** [comment]
This replaces https://github.com/Katello/katello/pull/6070

File: config/routes/api/v2.rb [hunk 1]
```diff
@@ -221,7 +225,10 @@ class ActionDispatch::Routing::Mapper
           end
         end
         api_resources :puppet_modules, :only => [:index, :show] do
-          get :auto_complete_search, :on => :collection
+          collection do
+            get :compare
```
Comment (ehelms) [inline config/routes/api/v2.rb]: I don't entirely follow. Looks like you already have code in place to handle that difference?

**Comment by johnpmitsch** [comment]
@ehelms This is finished and you can review. I haven't touched puppet modules yet as that is going to require larger changes as we discussed off-thread. Errata, docker tags, and packages should all be comparable in a CV via the API

**Comment by johnpmitsch** [comment]
closing this due to not actively working on it, and it also may require some larger changes in katello before this can be properly implimented

**Pull request closed without merging**